core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -270,9 +270,19 @@ private[spark] object JettyUtils extends Logging {
 
       gzipHandlers.foreach(collection.addHandler)
       connectors.foreach(_.setHost(hostName))
+      // As each Acceptor will use one thread, the number of threads should at least be the number
+      // of acceptors plus 1. (See SPARK-13776)
+      var minThreads = 1
+      connectors.collect { case c: AbstractConnector => c }.foreach { c =>
```

Comment:
Nit: the Scala math class is `math` vs `Math` but I wouldn't change it just for that. I think this looks like the right kind of approach.